test/overlapped-checker/main_unix.c

```diff
@@ -0,0 +1,35 @@
+#include <stdio.h>
+
+#include <unistd.h>
+
+static size_t r(char *buf, size_t buf_size) {
+  ssize_t read_count = read(0, buf, buf_size);
```

Comment:
This needs to be something like this:
```c
// #include <errno.h>
ssize_t read_count;
do
  read_count = read(0, buf, buf_size);
while (read_count == -1 && errno == EINTR);
```
Ditto for the `write()` call. Reads and writes can be interrupted by signals. That's benign but needs to be handled.